ofrak_patch_maker/ofrak_patch_maker/toolchain/utils.py

```diff
@@ -5,15 +5,19 @@
 from multiprocessing import Pool, cpu_count
 from typing import Optional, Dict, Mapping, Tuple
 
-import magic
-
 from ofrak_patch_maker.toolchain.model import BinFileType, Segment
 from ofrak_type.error import NotFoundError
 from ofrak_type.memory_permissions import MemoryPermissions
 
+_magic = None
+
 
 def get_file_format(path):
-    result = magic.from_file(path).split(" ")[0].lower()
+    global _magic
+    if _magic is None:
+        import magic as _magic
+
+    result = _magic.from_file(path).split(" ")[0].lower()
```

Comment:
Why is this change needed? I don't understand what is different.

Reply:
If it is left as a global import, it doesn't matter if a user (without libmagic installed) wants to exclude components missing dependencies because a hard error will be raised at import time when PatchMaker code is imported. The expected behavior is that as a user without an external tool dependency installed, I should still be able to launch and use OFRAK without needing to install all those tools, working instead with a limited set of components.

Since we only use magic in this one place, I thought it acceptable to sort of localize the magic import.